app/javascript/packs/service_discovery.js

```diff
@@ -0,0 +1,116 @@
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+
+document.addEventListener('DOMContentLoaded', () => {
+  const BASE_URL = '/p/admin/service_discovery/'
+  const formSource = document.getElementById('new_service_source')
+  const formScratch = document.querySelectorAll('form#new_service, form#create_service')[0]
+  const formDiscover = document.getElementById('service_discovery')
+
+  const clearSelectOptions = select => {
+    const options = select.getElementsByTagName('option')
+    for (let option of options) {
+      option.remove()
+    }
+  }
+
+  const clearNamespaces = () => clearSelectOptions(formDiscover.service_namespace)
+
+  const clearServiceNames = () => clearSelectOptions(formDiscover.service_name)
+
+  const showAndHideForms = (showForm, hideForm) => {
```

Comment:
I would extract 2 method of this.

Reply:
The idea is to show 1 form and hide the other. If we extract, we will have 4 calls instead of 2